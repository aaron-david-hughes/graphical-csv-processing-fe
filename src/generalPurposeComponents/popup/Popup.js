import React from 'react';
import TitleBox from "../titleBox/TitleBox";

class Popup extends React.Component {
    render() {
        return <div
            className='popup'
            style={{
                display: this.props.isOpen ? 'block' : 'none'
            }}
        >
            <div
                id='backgroundCover'
                onClick={this.props.close}
                style={{
                    width: '100%',
                    height: '100%',
                    background: 'rgba(0, 0, 0, .5)',
                    position: 'fixed',
                    top: '0',
                }}
            />

            <TitleBox
                id={this.props.id}
                title={this.props.title}
                className='popupSize'
                style={{
                    width: this.props.width,
                    height: this.props.height,
                    top: `calc(50% - 0.5 * ${this.props.height})`,
                    left: `calc(50% - 0.5 * ${this.props.width})`,
                    position: 'fixed'
                }}
                open={true}
                icon={
                    <i
                        className='fa fa-times'
                        style={{
                            fontSize: '24px',
                            color: 'darkslategray'
                        }}
                    />
                }
                iconOnClick={this.props.close}
                iconTitle='Close'
                fitContent={this.props.fitContent}
                scrollable={this.props.scrollable}
            >
                {this.props.children}
            </TitleBox>
        </div>
    }
}

export default Popup;