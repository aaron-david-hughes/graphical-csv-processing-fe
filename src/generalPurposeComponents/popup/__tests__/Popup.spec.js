import React from 'react';
import {shallow} from 'enzyme';
import Popup from "../Popup";

describe('Popup tests', () => {
    let props = {
        close: jest.fn(),
        isOpen: true,
        id: 'testPopupId',
        title: 'testPopupTitle',
        height: '10px',
        width: '10px',
        children: 'titleBoxContent'
    };

    let render;

    beforeEach(() => {
        render = shallow(<Popup {...props} />);
    });

    describe('Popup render tests', () => {
        it('matches the snapshot when open', () => {
            expect(render).toMatchSnapshot();
        });

        it('matches the snapshot when closed', () => {
            render = shallow(<Popup {...props} isOpen={false} />);

            expect(render).toMatchSnapshot();
        });

        it('should call the close function when click occurs elsewhere on DOM', () => {
            render.find('#backgroundCover').simulate('click');

            expect(props.close).toHaveBeenCalled();
        });
    });
});
