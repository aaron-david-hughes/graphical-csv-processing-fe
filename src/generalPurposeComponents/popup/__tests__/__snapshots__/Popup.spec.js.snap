// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popup tests Popup render tests matches the snapshot when closed 1`] = `
<div
  className="popup"
  style={
    Object {
      "display": "none",
    }
  }
>
  <div
    id="backgroundCover"
    onClick={[MockFunction]}
    style={
      Object {
        "background": "rgba(0, 0, 0, .5)",
        "height": "100%",
        "position": "fixed",
        "top": "0",
        "width": "100%",
      }
    }
  />
  <TitleBox
    icon={
      <i
        className="fa fa-times"
        style={
          Object {
            "color": "darkslategray",
            "fontSize": "24px",
          }
        }
      />
    }
    iconOnClick={[MockFunction]}
    iconTitle="Close"
    id="testPopupId"
    open={true}
    style={
      Object {
        "height": "10px",
        "left": "calc(50% - 0.5 * 10px)",
        "position": "fixed",
        "top": "calc(50% - 0.5 * 10px)",
        "width": "10px",
      }
    }
    title="testPopupTitle"
  >
    titleBoxContent
  </TitleBox>
</div>
`;

exports[`Popup tests Popup render tests matches the snapshot when open 1`] = `
<div
  className="popup"
  style={
    Object {
      "display": "block",
    }
  }
>
  <div
    id="backgroundCover"
    onClick={[MockFunction]}
    style={
      Object {
        "background": "rgba(0, 0, 0, .5)",
        "height": "100%",
        "position": "fixed",
        "top": "0",
        "width": "100%",
      }
    }
  />
  <TitleBox
    icon={
      <i
        className="fa fa-times"
        style={
          Object {
            "color": "darkslategray",
            "fontSize": "24px",
          }
        }
      />
    }
    iconOnClick={[MockFunction]}
    iconTitle="Close"
    id="testPopupId"
    open={true}
    style={
      Object {
        "height": "10px",
        "left": "calc(50% - 0.5 * 10px)",
        "position": "fixed",
        "top": "calc(50% - 0.5 * 10px)",
        "width": "10px",
      }
    }
    title="testPopupTitle"
  >
    titleBoxContent
  </TitleBox>
</div>
`;
